app/models/access.rb

```diff
@@ -0,0 +1,38 @@
+class Access < ApplicationRecord
+  ALLOWED_RESOURCE_TYPES = %w[Organization FacilityGroup Facility].freeze
+
+  belongs_to :user
+  belongs_to :resource, polymorphic: true, optional: true
+
+  enum role: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
+  }
+
+  validates :role, presence: true
+  validates :resource_type, inclusion: {in: ALLOWED_RESOURCE_TYPES}, allow_nil: true
+
+  class << self
+    def organizations
+      resources_for(Organization)
+    end
+
+    def facility_groups
+      resources_for(FacilityGroup)
+        .or(FacilityGroup.where(organization: resources_for(Organization)))
+    end
+
+    def facilities
+      resources_for(Facility)
+        .or(Facility.where(facility_group: FacilityGroup.where(organization: resources_for(Organization))))
+        .or(Facility.where(facility_group: resources_for(FacilityGroup)))
+    end
+
+    private
+
+    def resources_for(type)
```

Comment:
This is a really useful function, both now, and in the future, if we need to make exceptions to our current scoping system. It returns the actual resource object for an `Access` object.